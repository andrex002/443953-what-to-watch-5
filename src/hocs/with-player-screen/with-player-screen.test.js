import React, {createRef} from "react";
import renderer from "react-test-renderer";
import withPlayerScreen from "./with-player-screen";
import PropTypes from "prop-types";

const forwardedRef = createRef();

const MockComponent = (props) => {
  const {children} = props;

  return (
    <React.Fragment>
      {children}
    </React.Fragment>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withPlayerScreen(MockComponent);


it(`withPlayerScreen is rendered correctly`, () => {
  const tree = renderer.create((
    
    <MockComponentWrapped>
         <React.Fragment />
      </MockComponentWrapped>
  ), {
    createNodeMock(MockComponent) {
      const forwardedRef = createRef();
      
      return (
        <MockComponent forwardedRef={forwardedRef} />
      );
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
