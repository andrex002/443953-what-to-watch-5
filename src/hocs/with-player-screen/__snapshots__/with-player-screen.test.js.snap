// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withPlayerScreen is rendered correctly 1`] = `
<video
  className="player__video"
  poster="img/the-grand-budapest-hotel-poster.jpg"
  src="https://some-link"
/>
`;
